frontend/app_flowy/lib/plugins/board/presentation/toolbar/board_setting.dart

```diff
@@ -166,3 +135,48 @@ extension _GridSettingExtension on BoardSettingAction {
     }
   }
 }
+
+class BoardSettingListPopover extends StatefulWidget {
+  final BoardSettingContext settingContext;
+
+  const BoardSettingListPopover({
+    Key? key,
+    required this.settingContext,
+  }) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() => _BoardSettingListPopoverState();
+}
+
+class _BoardSettingListPopoverState extends State<BoardSettingListPopover> {
+  bool _showGridPropertyList = false;
```

Comment:
Seems like we created implicit dependencies here. We assume the BoardSettingListPopover must be used as a popover. Would it be better if we create a BoardSettingList and let the developer decide how to show it?

Reply:
I think there is no need to use the stateful widget. We can let the `popupBuilder` decide whether to rebuild it or not. Just like the `BlocBuilder` does. 

Also, I saw lots of stateless widgets are refactored to stateful widgets in order to keep the  `PopoverController`. So we can use   _popover.show(). But it seems very invasive.

I prefer something like context.read<PopoverController>.show(). Just like the `BlocProvider` does.

Reply:
Most of the time, the parent widget doesn't need to pass a `PopoverController`. They use `triggerActions` to control the popover.
The controller's style matched the style of flutter such as `InputController`.

And it can't be refactored into a `context.read<PopoverController>` because the `PopoverController` is held by a parent widget to open a child widget. It's not used in the popover child. If a child popover wants to close itself, it can call:

```
PopoverContainerState.of(popoverContext).closeAll();
```
Example: `app_flowy/lib/plugins/grid/presentation/widgets/header/type_option/date.dart`

Reply:
> Most of the time, the parent widget doesn't need to pass a `PopoverController`. They use `triggerActions` to control the popover. The controller's style matched the style of flutter such as `InputController`.
> 
> And it can't be refactored into a `context.read<PopoverController>` because the `PopoverController` is held by a parent widget to open a child widget. It's not used in the popover child. If a child popover wants to close itself, it can call:
> 
> ```
> PopoverContainerState.of(popoverContext).closeAll();
> ```
> 
> Example: `app_flowy/lib/plugins/grid/presentation/widgets/header/type_option/date.dart`

Got it.

Reply:
Would it be better if we use `PopoverContainer.of(context).closeAll()` because the state of PopoverContainer should be private

Reply:
Reasonable